dataprep/clean/clean_ip.py

```
@@ -0,0 +1,256 @@
+        Displays how many queries were removed from rows
+        Default value is set to `True`
+    errors
+        Specify ways to deal with broken value
+        {'ignore', 'coerce', 'raise'}, default 'coerce'
+        'raise': raise an exception when there is broken value
+        'coerce': set invalid value to NaN
+        'ignore': just return the initial input
+
+    Returns
+    ----------
+    A new Dataframe with the new relavant columns
+
+    """
+    reset_ip()
+    df = to_dask(df)
+
+    # specify the metadata for dask apply
+    meta = df.dtypes.to_dict()
+
+    meta[f"{column}_transformed"] = str
+
+    df = df.apply(format_ip, args=(column, input_format, output_format, errors), axis=1, meta=meta)
+
+    df, nrows = dask.compute(df, df.shape[0])
+
+    if inplace:
+        df = df.drop(columns=[column])
+
+    if report:
+        report_ip(nrows=nrows, errors=errors, column=column)
+
+    return df
+
+
+def format_ip(
+    row: pd.Series, column: str, input_format: str, output_format: str, errors: str
+) -> pd.Series:
+    """
+    This function formats each row of a pd.Series containing the url column
+    """
+    val_dict = transform_ip(
+        row[column],
+        input_format=input_format,
+        output_format=output_format,
+        errors=errors,
+    )
+    row[f"{column}_transformed"] = val_dict
+
+    return row
+
+
+def transform_ip(ip_address: str, input_format: str, output_format: str, errors: str) -> Any:
+    """
+    cleans and formats individual ip address entries
+    """
+
+    if check_ip(ip_address):
+        return convert_ip(
+            ip_address=ip_address, input_format=input_format, output_format=output_format
+        )
+
+    else:
+        if check_first():
+            STATS["first_val"] = 200
+        STATS["incorrect_format"] += 1
+        if errors == "coerce":
+            return np.nan
+        elif errors == "raise":
+            raise ValueError("Cannot parse this value" + ip_address)
+        else:
```

May be better to throw an exception if the input error type is not supported

---

Hi Yi, we are throwing an error by raising a `ValueError` for unparsable values.

---

What I mean is that we should throw an exception if the input for `errors` is not one of `raise`, `ignore` and `coerce`. Have we included that?

---

Got it, I have made the necessary changes to handle those errors.